Contributions/Linux_Memory_Management_Essentials.md

```diff
@@ -0,0 +1,249 @@
+6. Assertions about page "mobility", from different perspectives:
+   1. employment of physical pages - observing a physical memory page (e.g. a 4kB chunk aligned on a 4kB boundary) and how it is employed: what sort of content it might host, over time.
+      1. certain physical pages are put to use for a specific purpose, which is (almost) immutable, for the entire duration of the execution of the kernel:
+         1. kernel code from the monolithic portion (except for pages containing kernel init code, which are released after kernel init is completed)
+         2. kernel statically allocated data from the monolithic portion (except for pages containing kernel init data, which are released after kernel init is completed)
+         3. some kernel dynamically allocated data, used by the kernel itself, and never released, due to the nature of its use (e.g. object caches, persistent buffers, etc.)
+         4. memory used for loadable kernel modules (code, data) is tied to the permanence of the module in memory - this is typically stable through the entire duration of the execution. Some exceptions are modules loaded/unloaded as a consequence of certain peripherals (dis)appearing, e.g. USB ones.
+      2. other physical pages (most, actually, in a typical system) are available for runtime allocation/release cycles of multiple users:
+         1. transient kernel linear memory allocations (kmalloc / get_free_pages)
+         2. transient kernel virtually linear memory allocations (vmalloc for kernel address space)
+         3. user-space memory allocations (they are always virtually linear), which are by default transient and exposed to repurposing proactively done by the kernel (see below)
+   2. transitioning of a logical page - given certain context and content, where it might be located in memory over time - and if it might be even discarded.
+      1. the kernel doesn't spontaneously repurpose the utilisation of its own physical pages; therefore it is possible to assume that the logical content of kernel allocations will reamin tied to the associated physical pages, as long as it is not intentionally altered (or subject to interference)
+         1. metadata memory used by the kernel for housekeeping purposes related to processes is included in this category; examples: task/cred structures, vmas, maple leaves, process page tables.
+      2. memory used by the kernel for the actual userspace processes: the content of this logical page determines its life cycle and expectancy: certain content such as code or constants can be "re-constructed" by relaoding it from file (code pages are likely to necessitate undergoing re-linking), so the actual logical content might disappear, over time. Other pages, on the other hand, are meant to hold non-re-constructible content, such as stack, heap, and variable data. These pages can, at most, be swapped out, and loaded back later on, but they cannot be simply dropped.
```

Comment:
can you better explain what you mean by "the actual logical content might disappear, over time" ?

Reply:
"live content" ? 
The content as it is intended to be in memory when the process needs to use it.

Reply:
Ah ok I see...you meant to say that "this type of content is safe to disappear, over time" (since it can be reconstructed) ?

Reply:
no, I was actually pointing to the risk associated with it disappearing and possiby not coming back correctly and/or timely XD 

but I could say oemthing like "by design this content miht disappear, and is expected to be reconstructed if/when needed" and then point to the associated risk

Reply:
I think it helps splitting this paragraph into the two separate cases: a) reconstructible and b) live or non-reconstructible pages

Reply:
+1 for breaking this up into two points

Reply:
I think the division done by @dweingaertner is not correct; it is all somehow reconstructible - if one allows for swap. Preventing swap simply means that only what is in theory reconstructible gets dropped. But the problems with the reconstructions apply.

Reply:
explained in the amended text